pkg/version/compatibility.go

```diff
@@ -62,3 +73,45 @@ var CompatibilityMatrix = Compatibility{
 }
 
 var DevVersionPrefix = "dev"
+
+// DeploymentVersionCompatible returns true if the given component version
+// is compatible with the installed version of Kuma CP.
+// For all binaries which share a common version (Kuma DP, CP, Zone CP...), we
+// support backwards compatibility of at most two prior minor versions.
+func DeploymentVersionCompatible(kumaVersionStr string, componentVersionStr string) bool {
+	if strings.Contains(kumaVersionStr, "dev") || strings.Contains(componentVersionStr, "dev") {
+		return true
+	}
+
+	kumaVersion, err := semver.NewVersion(kumaVersionStr)
+	if err != nil {
+		// Assume some kind of dev version
+		log.Info("cannot parse semantic version", "version", kumaVersionStr)
+		return true
+	}
+
+	componentVersion, err := semver.NewVersion(componentVersionStr)
+	if err != nil {
+		// Assume some kind of dev version
+		log.Info("cannot parse semantic version", "version", componentVersionStr)
+		return true
+	}
+
+	minMinor := int64(kumaVersion.Minor()) - 2
+	if minMinor < 0 {
+		minMinor = 0
+	}
```

Comment:
I took this but I'm actually going to revert it because `math.Max()` requires `float` and I don't want to potentially hit some conversion corner case jumping between integers / floats / semver strings.